test/e2e/lib/gutenberg/tracking/general-tests.js

```diff
@@ -158,6 +158,54 @@ export function createGeneralTests( { it, editorType, postType } ) {
 		);
 	} );
 
+	it( 'Tracks "wpcom_pattern_inserted"', async function () {
+		const editor = await EditorComponent.Expect( this.driver, gutenbergEditorType );
+
+		await editor.insertPattern( 'list', 'List with Image' );
+		const eventsStackList = await getEventsStack( this.driver );
+		await clearEventsStack( this.driver );
+
+		await editor.insertPattern( 'gallery', 'Heading and Three Images' );
+		// We need to save the eventsStack after each insertion to make sure we
+		// aren't running out of the E2E queue size.
+		const eventsStackGallery = await getEventsStack( this.driver );
```

Comment:
Is this still necessary after increasing the stack size?

Reply:
The reason we had to increase the stack size is that inserting the `Heading and Three Images` pattern exhausted the stack size on its own with `wpcom_block_inserted` events.

Though the clearEventsStack is not necessary anymore since the queue size is big enough to hold all the events, I'd lean towards keeping it to be on the safe side. What do you think?

Reply:
Sounds fair.